services/galley/src/Galley/API/Message.hs

```diff
@@ -483,8 +486,8 @@ postQualifiedOtrMessage senderType sender mconn lcnv msg =
           redundant' = toDomUserClient $ qualifiedUserClients $ mssRedundantClients otrResult
           -- List of users that we couldn't fetch clients for. Used to get their "redundant"
           -- clients for reporting as failedToSend.
-          failed' = toDomUserClient $ qualifiedUserClients failedToSendFetchingClients
-          -- failedToSendFetchingClients doesn't contain client IDs, so those need to be excluded
+          failed' = toDomUserClient $ toDomMap unconfirmedUnknownClients
+          -- failedToConfirmRemoteClients doesn't contain client IDs, so those need to be excluded
           -- from the filter search. We have to focus on only the domain and user. These clients
           -- should be listed in the failedToSend field however, as tracking these clients is an
```

Comment:
wait, does that mean we're still reporting clients that will eventually get the message in "failed to send"?  i understood this shouldn't be, as we expect the clients to be reached eventually?

Reply:
never mind, the comment is talking about clients to be added to "failed" because we couldn't reach the backend to ask what clients this user owns.  (these clients can be recoved using the user id from the send-msg response and the client ids from the send-msg request.)

it was all in my head!  :)